include/mbedtls/build_info.h

```diff
@@ -87,6 +87,12 @@
 #define MBEDTLS_MD_C
 #endif
 
+/* PSA crypto specific configuration options */
```

Comment:
I think this placement is correct now. But it needs some explanation: what must be above/below the inclusion of `config_psa.h`? What can `config_psa.h` read/write?

Does the following rule work?

* `config_psa.h` does not read or write `MBEDTLS_xxx_LIGHT`, so these symbols can be enabled at any time.
* `config_psa.h` reads and does not write `MBEDTLS_MD_C`, so this symbol can be enabled before its inclusion, but not after.
* `config_psa.h` writes and does not read `MBEDTLS_PK_xxx`, so these symbols can be consulted after its inclusion, but not before, and they can be enabled before or after.
* `config_psa.h` reads and writes symbols directly related to a specific mechanism (`MBEDTLS_ECP_C`, `MBEDTLS_RSA_C`, etc.) so `build_info.h` must not set these symbols, and can only consult them after.

Reply:
Thank you Gilles. It seems the first paragraph about `MBEDTLS_xxx_LIGHT` is not accurate.

Assume it's defined as this way in `build_info.h`
```
#if defined(AAA)
#define BBB
#endif
```
I think we should consider how `AAA` and `BBB` are impacted by `config_psa.h` together. How about writing it as below?

* If `config_psa.h` writes `AAA`, this symbol should be consulted after its inclusion.
* If `config_psa.h` reads `BBB`, this symbol should be consulted before its inclusion.
* As for other cases, `AAA` can be enabled at any time. But it's recommended to enable the symbol after inclusion.

Does this make sense?

Reply:
I simplified the explanation a little bit. It looks we only have to write when the symbol should be defined before inclusion of `config_psa.h` and when the symbol should be defined after inclusion of `config_psa.h`. See new commit 419a55e